File: 55. Jump Game/55.py
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        max_jump = 0
        end = len(nums) -1

        for i, jump in enumerate(nums):
            if i > max_jump:
                return False

            max_jump = max(max_jump, i + jump)

            if max_jump >= end:
                return True


        



S = Solution()
s = S.canJump([1,1,1,0])
print(s)

File: 55. Jump Game/55.cs
using System.Diagnostics;

public class Solution {
    public bool CanJump(int[] nums) {
        int max_jump = 0;
        int end = nums.Count() - 1;

        for (int i = 0; i <= end; i++)
        {
            if (i > max_jump)
            {
                return false;
            }
            max_jump = Math.Max(max_jump, i+nums[i]);

            if (max_jump >= end)
            {
                return true;
            }
        }

        return true;
        
    }
}